packages/core/src/components/DatePicker/DatePicker.styled.tsx

```diff
@@ -1,7 +1,47 @@
-import { styled } from '@medly-components/utils';
-import { default as MedlyInput } from '../Input';
+import { SvgIcon } from '@medly-components/icons';
+import { css, styled, WithThemeProp } from '@medly-components/utils';
+import { OuterWrapper } from '../TextField/Styled';
+import { InnerWrapper } from '../TextField/Styled/InnerWrapper.styled';
+import { StyleProps } from './types';
 
-export const Input = styled(MedlyInput)`
-    user-select: none;
-    cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'inherit')};
+type State = 'default' | 'active' | 'error' | 'disabled';
+
+const getStyleForIcon = ({ theme, variant }: StyleProps & WithThemeProp, state: State) => {
+    const { datePicker } = theme;
+    const {
+        icon: { [state]: iconStyle }
+    } = datePicker[variant];
+    return css`
+        ${SvgIcon} {
+            border-radius: 50%;
+            padding: 0.8rem;
+            * {
+                fill: ${iconStyle.color};
+            }
+            &:hover {
+                background-color: ${iconStyle.backgroundColor};
+                * {
+                    fill: ${iconStyle.hoverColor};
+                }
+            }
+        }
+    `;
+};
+
+export const DateIcon = styled.span<StyleProps>`
+    height: 4rem;
+    ${props => getStyleForIcon(props, 'default')};
+    ${props => props.active && getStyleForIcon(props, 'active')};
+    ${props => props.errorText && getStyleForIcon(props, 'error')};
+    ${props => props.disabled && getStyleForIcon(props, 'disabled')};
+`;
+
+export const Wrapper = styled.div<StyleProps>`
+    ${InnerWrapper} {
```

Comment:
why we need this ?

Reply:
This is needed because the **icon** has its own padding and rounded background, with even spacing icon seemed to be shifting to right. 
Although negative margin was in mind, but thought this could be better approach.